crates/server/src/scheduled.rs

```diff
@@ -14,69 +14,91 @@
+        }
     }
 
     async fn do_report(&self) {
         let metrics = GlobalMetrics::get();
-        let engine = &self.ctx.engine;
 
-        metrics.storage.capacity.record(engine.capacity(), &[]);
-        // Foyer will reserve all the space in the disk, so the used space is meaningless
+        let engine = &self.ctx.engine;
+        // foyer will reserve all the space in the disk; the used space is meaningless
         metrics.storage.used.record(engine.capacity(), &[]);
+        metrics.storage.capacity.record(engine.capacity(), &[]);
 
-        let stats = engine.stats();
         let read_label = StorageIOMetrics::operation_labels(StorageIOMetrics::OPERATION_READ);
         let write_label = StorageIOMetrics::operation_labels(StorageIOMetrics::OPERATION_WRITE);
-        let flush_label = StorageIOMetrics::operation_labels(StorageIOMetrics::OPERATION_FLUSH);
-        metrics.storage.io.bytes.add(
-            stats.read_bytes.load(std::sync::atomic::Ordering::Relaxed) as u64,
-            &read_label,
-        );
-        metrics.storage.io.bytes.add(
-            stats.write_bytes.load(std::sync::atomic::Ordering::Relaxed) as u64,
-            &write_label,
-        );
-        metrics.storage.io.count.add(
-            stats.read_ios.load(std::sync::atomic::Ordering::Relaxed) as u64,
-            &read_label,
-        );
-        metrics.storage.io.count.add(
-            stats.write_ios.load(std::sync::atomic::Ordering::Relaxed) as u64,
-            &write_label,
-        );
-        metrics.storage.io.count.add(
-            stats.flush_ios.load(std::sync::atomic::Ordering::Relaxed) as u64,
-            &flush_label,
-        );
 
-        // Reset the stats
-        stats
-            .read_ios
-            .store(0, std::sync::atomic::Ordering::Relaxed);
-        stats
-            .read_bytes
-            .store(0, std::sync::atomic::Ordering::Relaxed);
-        stats
-            .write_ios
-            .store(0, std::sync::atomic::Ordering::Relaxed);
-        stats
-            .write_bytes
-            .store(0, std::sync::atomic::Ordering::Relaxed);
-        stats
-            .flush_ios
-            .store(0, std::sync::atomic::Ordering::Relaxed);
+        let current = MetricsSnapshot::from(engine.statistics().as_ref());
+        let previous = self.snapshot.load();
+        let difference = current.difference(&previous);
+        self.snapshot.store(Arc::new(current));
+
+        metrics
+            .storage
+            .io
+            .bytes
+            .add(difference.disk_read_bytes, &read_label);
+        metrics
+            .storage
+            .io
+            .bytes
+            .add(difference.disk_write_bytes, &write_label);
+        metrics
+            .storage
+            .io
+            .count
+            .add(difference.disk_read_ios, &read_label);
+        metrics
+            .storage
+            .io
+            .count
+            .add(difference.disk_write_ios, &write_label);
```

Comment:
@MrCroxx BTW, what does "ios" stand for? Is it guaranteed to increase monotonically?

Reply:
Otherwise, doing difference can cause usize/u64 underflow.

Reply:
Yes. It counts the IO request count. (To adopt to AWS EBS IO request count rule, it can be configured to count IO request by aligning to 128KiB.